x/twap/logic_test.go

```diff
@@ -42,28 +43,43 @@ func (s *TestSuite) TestGetSpotPrices() {
 		expectedSp1           sdk.Dec
 		expectedLatestErrTime time.Time
 	}{
-		"zero sp": {
+		"use last spot price on err": {
 			poolID:                poolID,
 			prevErrTime:           currTime,
+			blockTime:             s.Ctx.BlockTime(),
 			mockSp0:               sdk.ZeroDec(),
 			mockSp1:               sdk.ZeroDec(),
 			mockSp0Err:            fmt.Errorf("foo"),
-			expectedSp0:           sdk.ZeroDec(),
+			expectedSp0:           sdk.OneDec(),
 			expectedSp1:           sdk.ZeroDec(),
-			expectedLatestErrTime: ctx.BlockTime(),
+			expectedLatestErrTime: s.Ctx.BlockTime(),
+		},
+		// no pre records at block t + 1
+		"sp zero, when err & no previous record": {
+			poolID:                poolID,
+			prevErrTime:           currTime,
+			blockTime:             currTime.Add(5 * time.Second),
+			mockSp0:               sdk.OneDec(),
+			mockSp1:               sdk.OneDec(),
+			mockSp0Err:            fmt.Errorf("foo"),
+			expectedSp0:           sdk.ZeroDec(),
+			expectedSp1:           sdk.OneDec(),
+			expectedLatestErrTime: currTime.Add(5 * time.Second),
```

Comment:
same question here, why does expected SP0 result in zerodec and expected sp1 result in one dec?

Reply:
`expectedSp0` = 0 cause got err when getting sp0 & no previous record (`blockTime` = t + 1 while pool create at block t + 2)

Reply:
Sorry if im missing anything, we still have the previous record for sp0 which is onedec right?

Reply:
At block `t + 1` we dont have any record